packages/web-components/@groupby/elements-sayt/src/sayt.ts

```diff
@@ -243,7 +243,7 @@ export default class Sayt extends Base {
    * @param event The KeyboardEvent object.
    */
   updateSearchboxInputTerm(event: KeyboardEvent): void {
-    if (event.key !== 'Enter' || !this.nodeInSearchbox(event.target as Node)) return;
+    if (event.key !== 'Enter' || !this.nodeInSearchbox(event.target as Node) || !this.visible) return;
```

Comment:
The least expensive operation should go first to take advantage of short-circuiting. I would first check `visible`, then `key`, then `nodeInSearchbox`. That way, the runtime doesn't have to check the DOM tree if it doesn't have to.